### PR TITLE
Refs #17209 -- Fixed token verification for PasswordResetConfirmView POST requests.

### PR DESCRIPTION
**Pull request by apollo13**

### CODE REVIEW

File: django/contrib/auth/views.py [hunk 1]
```diff
@@ -441,6 +441,15 @@ class PasswordResetConfirmView(PasswordContextMixin, FormView):
...
     def dispatch(self, *args, **kwargs):
         assert 'uidb64' in kwargs and 'token' in kwargs
+
+        self.validlink = False
+        self.user = self.get_user(kwargs['uidb64'])
+
+        if self.user is not None and self.token_generator.check_token(self.user, kwargs['token']):
+            self.validlink = True
+        else:
+            return self.render_to_response(self.get_context_data())
```
Comment (auvipy) [inline django/contrib/auth/views.py]: should't this be render() ?

File: tests/auth_tests/test_views.py [hunk 2]
```diff
@@ -253,7 +253,23 @@ def test_confirm_invalid_post(self):
...
+        url, path = self._test_confirm_start()
+        path_parts = path.split('-')
+        path_parts[-1] = ("0") * 20 + '/'
+        path = '-'.join(path_parts)
+
+        self.client.post(path, {
+            'new_password1': 'anewpassword',
+            'new_password2': 'anewpassword',
+        })
+        # Check the password has not been changed
```
Comment (timgraham) [inline tests/auth_tests/test_views.py]: I think `self.assertIs(response.context['validlink'], False)` would work?

Comment (MarkusH) [inline tests/auth_tests/test_views.py]: I'd rather be explicit here and refresh from the database. On that note, there's `u.refresh_from_db()` :wink:

Comment (timgraham) [inline tests/auth_tests/test_views.py]: pushed the cosmetic edits

**Pull request merged**